operators/pkg/controller/elasticsearch/elasticsearch_controller.go

```diff
@@ -87,29 +90,36 @@ func addWatches(c controller.Controller, r *ReconcileElasticsearch) error {
 		return err
 	}
 
-	// Watch pods
-	if err := c.Watch(&source.Kind{Type: &corev1.Pod{}}, r.dynamicWatches.Pods); err != nil {
+	// Watch StatefulSets
+	if err := c.Watch(
+		&source.Kind{Type: &appsv1.StatefulSet{}}, &handler.EnqueueRequestForOwner{
+			IsController: true,
+			OwnerType:    &elasticsearchv1alpha1.Elasticsearch{},
+		},
+	); err != nil {
 		return err
 	}
-	if err := r.dynamicWatches.Pods.AddHandlers(
-		// trigger reconciliation loop on ES pods owned by this controller
-		&watches.OwnerWatch{
-			EnqueueRequestForOwner: handler.EnqueueRequestForOwner{
-				IsController: true,
-				OwnerType:    &elasticsearchv1alpha1.Elasticsearch{},
-			},
-		},
-		// Reconcile pods expectations.
-		// This does not technically need to be part of a dynamic watch, since it will
-		// stay there forever (nothing dynamic here).
-		// Turns out our dynamic watch mechanism happens to be a pretty nice way to
-		// setup multiple "static" handlers for a single watch.
-		watches.NewExpectationsWatch(
-			"pods-expectations",
-			r.podsExpectations,
-			// retrieve cluster name from pod labels
-			label.ClusterFromResourceLabels,
-		)); err != nil {
+
+	// Watch pods belonging to ES clusters
```

Comment:
Do we need to watch pods now?

Reply:
Yes: we need to reconcile certificates once pods have an IP. Pods suddenly getting an IP in the status does not trigger any changes in the StatefulSet status :(
Not doing that leads to cluster getting stuck at bootstrap (reconciliation needs to happen for the init container waiting for certs, but does not happen).